### PR TITLE
Add redirect to spotify when opening the application

### PR DESCRIPTION
**Pull request by 0DarthSanta0**

### CODE REVIEW

File: app/src/main/java/com/example/music_app/ui/screens/login/LoginScreen.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.viewinterop.AndroidView
+import com.example.music_app.AUTH_URL
+import com.example.music_app.ui.models.classes.AppLoginWebViewClient
+
+
+@Composable
+fun LoginScreen() {
+
+    AndroidView(factory = {
```
Comment (valeriy3x) [inline app/src/main/java/com/example/music_app/ui/screens/login/LoginScreen.kt]: Please use Accompanist WebView for Jetpack Compose (https://google.github.io/accompanist/api/web/com.google.accompanist.web/-web-view.html). Avoid using `AndroidView` inside of the Composables.

File: app/build.gradle [hunk 2]
```diff
@@ -57,6 +57,9 @@ dependencies {
     implementation 'androidx.compose.ui:ui-graphics'
     implementation 'androidx.compose.ui:ui-tooling-preview'
     implementation 'androidx.compose.material3:material3'
+    implementation 'androidx.appcompat:appcompat:1.4.1'
+    implementation 'com.google.android.material:material:1.5.0'
+    implementation 'androidx.constraintlayout:constraintlayout:2.1.3'
```
Comment (valeriy3x) [inline app/build.gradle]: Why do you need `constraintlayout` dependency? If it's not needed, could you please remove it

File: app/src/main/java/com/example/music_app/MainActivity.kt [hunk 3]
```diff
@@ -3,44 +3,16 @@ package com.example.music_app
...
-            MusicappTheme {
-                // A surface container using the 'background' color from the theme
-                Surface(
-                    modifier = Modifier.fillMaxSize(),
-                    color = MaterialTheme.colorScheme.background
-                ) {
-                    Greeting("Android")
-                }
-            }
+            LoginScreen()
```
Comment (valeriy3x) [inline app/src/main/java/com/example/music_app/MainActivity.kt]: Please add `MusicappTheme` wrapper back

File: app/src/main/java/com/example/music_app/ui/models/classes/AppLoginWebViewClient.kt [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+import android.net.Uri
+import android.webkit.WebView
+import android.webkit.WebViewClient
+
+class AppLoginWebViewClient : WebViewClient() {
+    override fun onPageFinished(view: WebView?, url: String?) {
+        super.onPageFinished(view, url)
+        val uri = Uri.parse(url)
+        val state = uri.getQueryParameter("state")
```
Comment (valeriy3x) [inline app/src/main/java/com/example/music_app/ui/models/classes/AppLoginWebViewClient.kt]: Please extract `state` string to the constant. Applies to `error` and `code` strings also

File: app/src/main/java/com/example/music_app/ui/models/classes/AppLoginWebViewClient.kt [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.example.music_app.ui.models.classes
+
+import android.net.Uri
+import android.webkit.WebView
+import android.webkit.WebViewClient
+
+class AppLoginWebViewClient : WebViewClient() {
+    override fun onPageFinished(view: WebView?, url: String?) {
+        super.onPageFinished(view, url)
+        val uri = Uri.parse(url)
```
Comment (valeriy3x) [inline app/src/main/java/com/example/music_app/ui/models/classes/AppLoginWebViewClient.kt]: Please move URL parsing logic to the separate method

File: app/src/main/java/com/example/music_app/ui/models/classes/AppLoginWebViewClient.kt [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+        val state = uri.getQueryParameter("state")
+        val error = uri.getQueryParameter("error")
```
Comment (valeriy3x) [inline app/src/main/java/com/example/music_app/ui/models/classes/AppLoginWebViewClient.kt]: Inside of the separate method query parameter extracting could be simplified with using `with`
```suggestion
        with(uri) {
        ....
        val error = getQueryParameter("error")
        ....
        }
```

File: app/build.gradle [hunk 7]
```diff
@@ -57,6 +57,9 @@ dependencies {
...
     testImplementation 'junit:junit:4.13.2'
```
Comment (valeriy3x) [inline app/build.gradle]: Please remove test dependencies

**Review by valeriy3x — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/music_app/ui/screens/login/LoginScreen.kt [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+
+@Composable
+fun LoginScreen() {
+    val state = rememberWebViewState(AUTH_URL)
+    WebView(
+        state,
+        onCreated = { webView ->
+            webView.settings.javaScriptEnabled = true
+            webView.webViewClient = AppLoginWebViewClient()
```
Comment (valeriy3x) [inline app/src/main/java/com/example/music_app/ui/screens/login/LoginScreen.kt]: Please use `with` function here

File: app/src/main/java/com/example/music_app/ui/screens/login/LoginScreen.kt [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+        }
+    )
+
```
Comment (valeriy3x) [inline app/src/main/java/com/example/music_app/ui/screens/login/LoginScreen.kt]: `Minor` please remove unnecessary empty lines here

File: app/src/main/java/com/example/music_app/constants/AuthUrl.kt [hunk 10]
```diff
@@ -0,0 +1,11 @@
+package com.example.music_app.constants
+
+const val SPOTIFY_CLIENT_ID = "67f6367abcd3497db54043591d9e7c9f"
+const val CODE = "code"
+const val SCOPE = "user-read-private user-read-email"
+const val REDIRECT_URL = "http://localhost:3000"
+
+const val STATE = "state"
+const val ERROR = "error"
```
Comment (valeriy3x) [inline app/src/main/java/com/example/music_app/constants/AuthUrl.kt]: `Suggestion` I would recommend placing this constants directly inside of the file with `AppLoginWebViewClient ` and make them private

File: app/build.gradle [hunk 11]
```diff
@@ -57,11 +57,14 @@ dependencies {
...
     implementation 'androidx.compose.ui:ui-tooling-preview'
     implementation 'androidx.compose.material3:material3'
-    testImplementation 'junit:junit:4.13.2'
+    implementation 'androidx.appcompat:appcompat:1.4.1'
+    implementation 'com.google.android.material:material:1.5.0'
+    implementation "com.google.accompanist:accompanist-webview:0.31.5-beta"
     androidTestImplementation 'androidx.test.ext:junit:1.1.5'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.5.1'
     androidTestImplementation platform('androidx.compose:compose-bom:2022.10.00')
     androidTestImplementation 'androidx.compose.ui:ui-test-junit4'
```
Comment (valeriy3x) [inline app/build.gradle]: Could you please remove all test dependencies

File: app/src/main/java/com/example/music_app/constants/AuthUrl.kt (hunk 10 above)
Comment (0DarthSanta0) [reply app/src/main/java/com/example/music_app/constants/AuthUrl.kt]: The point of placing these constants in a common file for url is that they may be needed elsewhere in the program. They are connected with the url so I placed them in the AuthUrl file

**Review by valeriy3x — APPROVED** [review]

**Pull request merged**